models/ape_store/base/ape_store_base_trades.sql

```diff
@@ -0,0 +1,104 @@
+{{ config(
+    alias = 'trades',
+    schema = 'ape_store_base',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['blockchain', 'tx_hash', 'tx_index'],
+    post_hook = '{{ expose_spells(
+                        blockchains = \'["base"]\',
+                        spell_type = "project",
+                        spell_name = "ape_store",
+                        contributors = \'["whale_hunter"]\') }}'
+   )
+}}
+
+{% set project_start_date = '2024-04-04' %}
+{% set blockchain = 'base' %}
+{% set weth_contract_address = '0x4200000000000000000000000000000000000006' %}
+
+with
+    deployments as (
+        select
+            evt_block_time,
+            'base' as blockchain,
+            token,
+            id,
+            evt_tx_from as deployer,
+            evt_tx_hash
+        from {{ source("ape_store_base", "Router_evt_CreateToken") }}
+    ),
+    bondingcurvetrades as (
+        select
+            evt_block_time as block_time,
+            date_trunc('day', evt_block_time) as block_date,
+            date_trunc('month', evt_block_time) as block_month,
+            '{{blockchain}}' as blockchain,
+            'Bonding Curve' as platform,
+            if(amount0out > 0, 'Buy', 'Sell') as type,
+            (amount1in + amount1out) / 1e18 * price as amount_usd,
+            (amount0out + amount1out) / 1e18 as token_bought_amount,
+            if(amount0out > 0, token.symbol, 'WETH') token_bought_symbol,
+            if(
+                amount0out > 0, token, {{ weth_contract_address }}
+            ) as token_bought_address,
+            (amount0in + amount1in) / 1e18 as token_sold_amount,
+            if(amount0out > 0, 'WETH', token.symbol) as token_sold_symbol,
+            if(
+                amount0out > 0, {{ weth_contract_address }}, token
+            ) as token_sold_address,
+            sender as user,
+            evt_tx_hash as tx_hash,
+            evt_index as tx_index
+        from {{ source('ape_store_base', 'Router_evt_swap') }} as swaps
+        left join
+            {{ source('tokens', 'erc20') }} as token
+            on (
+                token.blockchain = '{{blockchain}}'
+                and token.contract_address = swaps.token
+            )
+        left join
+            {{ source('prices', 'usd') }} as ethPrice
+            on (
```

Comment:
add join condition to filter incrementally

Reply:
this should not be necessary, since in the join condition we already filter time right?

```sql
left join
            {{ source('prices', 'usd') }} as ethPrice
            on (
                ethPrice.blockchain = '{{blockchain}}'
                and ethPrice.contract_address = {{ weth_contract_address }}
                and ethPrice.minute = date_trunc('minute', evt_block_time)
            )
```

Reply:
it's still necessary, as trino will scan the entire prices dataset on left joins in order to match on `minute`. prices is quite large. there's no reason to try and read more data than our base source. plz do apply to ensure proper planning by the trino engine